libr/core/cmd_anal.c

```diff
@@ -149,6 +152,54 @@ static int var_cmd (RCore *core, const char *str) {
 			}
 		}
 		break;
+	case 'd': {
+		char *name = r_str_chop(strdup(str+2));
+		RAnalVar *v = r_anal_var_get_byname (core->anal, fcn, type, name);
+		if (!v) {
+			eprintf ("no arg/local with this name exists\n");
+			free (name);
+			break;
+		}
+		char *fmt = r_anal_type_format (core->anal, v->type);
+		if (!fmt) {
+			 eprintf ("type:%s doesn't exist\n", v->type);
+			 break;
+		}
+		switch (type) {
+		case 'r':{
+				RRegItem *i = r_reg_index_get (core->anal->reg, v->delta);
+				if (!i) {
```

Comment:
better use positive conditionals and 'i' is meant to be used for iterators. maybe `ri` is more adequate.

so do something like if (ri) { r_core_cmdf (core, "pf r ...\n"); } else { eprintf ("register not found"); }